src/diskio/mod.rs

```diff
@@ -160,29 +160,30 @@ pub fn write_file<P: AsRef<Path>, C: AsRef<[u8]>>(
     contents: C,
     mode: u32,
 ) -> io::Result<()> {
-    let mut opts = OpenOptions::new();
-    #[cfg(unix)]
-    {
-        use std::os::unix::fs::OpenOptionsExt;
-        opts.mode(mode);
-    }
-    let path = path.as_ref();
-    let path_display = format!("{}", path.display());
-    let mut f = {
-        trace_scoped!("creat", "name": path_display);
-        opts.write(true).create(true).truncate(true).open(path)?
-    };
-    let contents = contents.as_ref();
-    let len = contents.len();
-    {
-        trace_scoped!("write", "name": path_display, "len": len);
-        f.write_all(contents)?;
-    }
-    {
-        trace_scoped!("close", "name:": path_display);
-        drop(f);
+    fn inner(path: &Path, contents: &[u8], mode: u32) -> io::Result<()> {
+        let mut opts = OpenOptions::new();
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::OpenOptionsExt;
+            opts.mode(mode);
+        }
+        let path_display = path.display().to_string();
+        let mut f = {
+            trace_scoped!("creat", "name": path_display);
+            opts.write(true).create(true).truncate(true).open(path)?
+        };
+        {
+            let len = contents.len();
+            trace_scoped!("write", "name": path_display, "len": len);
+            f.write_all(contents)?;
+        }
+        {
+            trace_scoped!("close", "name:": path_display);
+            drop(f);
+        }
+        Ok(())
     }
-    Ok(())
+    inner(path.as_ref(), contents.as_ref(), mode)
```

Comment:
This rework seems unrelated to the commit it's in, and not really related to this PR at all.  is it left over from other work you were doing?

Reply:
Yeah. It is unrelated. I will split it to separate PR.

Reply:
Done.